plugins/azure_sentinel/icon_azure_sentinel/util/api.py

```diff
@@ -259,3 +266,401 @@ def list_entities(
+        final_uri = uri.format(
+            subscription_id,
+            resource_group_name,
+            workspace_name,
+            watchlist_alias,
+            watchlist_item_id,
+            api_version,
+        )
+        data = kwargs
+        data = return_non_empty(data)
+        _, result = self._call_api("PUT", final_uri, headers=self.headers, payload=data)
+        return result
+
+    def get_watchlist_items(
+        self,
+        resource_group_name: str,
+        workspace_name: str,
+        subscription_id: str,
+        watchlist_alias: str,
+        watchlist_item_id: str,
+        api_version: str = "2021-10-01",
+    ):
+        uri = Endpoint.GET_WATCHLIST_ITEMS
+        final_uri = uri.format(
+            subscription_id,
+            resource_group_name,
+            workspace_name,
+            watchlist_alias,
+            watchlist_item_id,
+            api_version,
+        )
+        _, result = self._call_api("GET", final_uri, headers=self.headers)
+        return result
+
+    def delete_watchlist_items(
+        self,
+        resource_group_name: str,
+        workspace_name: str,
+        subscription_id: str,
+        watchlist_alias: str,
+        watchlist_item_id: str,
+        api_version: str = "2021-10-01",
+    ):
+        uri = Endpoint.DELETE_WATCHLIST_ITEMS
+        final_uri = uri.format(
+            subscription_id,
+            resource_group_name,
+            workspace_name,
+            watchlist_alias,
+            watchlist_item_id,
+            api_version,
+        )
+        _, result = self._call_api("DELETE", final_uri, headers=self.headers)
+        return result
+
+    def list_watchlist_items(
+        self,
+        resource_group_name: str,
+        workspace_name: str,
+        subscription_id: str,
+        watchlist_alias: str,
+        api_version: str = "2021-10-01",
+    ):
+        uri = Endpoint.LIST_WATCHLIST_ITEMS
+        final_uri = uri.format(
+            subscription_id,
+            resource_group_name,
+            workspace_name,
+            watchlist_alias,
+            api_version,
+        )
+        _, result = self._call_api("GET", final_uri, headers=self.headers)
+        return result
+
+    def _prepared_necessary_data(self, indicator, data):
+        data["etag"] = indicator.get("etag")
+        data["kind"] = indicator.get("kind")
+        # optional data
+        if not data["properties"].get("displayName"):
+            data["properties"]["displayName"] = indicator["properties"].get("displayName")
+        if not data["properties"].get("pattern"):
+            data["properties"]["pattern"] = indicator["properties"].get("pattern")
+        if not data["properties"].get("patternType"):
+            data["properties"]["patternType"] = indicator["properties"].get("patternType")
+        if not data["properties"].get("threatTypes"):
+            data["properties"]["threatTypes"] = indicator["properties"].get("threatTypes")
```

Comment:
Would it be better to use `for loop` in here? What do you think?
```suggestion
        necessary_keys = ["displayName", "pattern", "patternType", "threatTypes"]
        for key in necessary_keys:
            if not data["properties"].get(key):
                data["properties"][key] = indicator["properties"].get(key)
```